### ISSUE TITLE
Hard targeting when the target is hidig "under" behind tower.

### ISSUE THREAD

**Issue by xAzta** [IroncladGames/SinsOfADarkAge#861]
Game ID: 151157
Around 15:20 we dived the enemy golem under tower, i played Ermuk, and it took me time to put cripple on the enemy or cast other spell on it.
Maybe my mouse was just in the wrong position, but if the enemy is totally hided behind the tower it's hard to see.

**Comment by writor** [comment]
Ah, the latest patch (42555) provided a workaround for this issue! The relevant notes are below.

Unit Targeting Modes
Implemented "Target Heroes Only" and "Target Non-Hero units Only" unit targeting modes to control what type of units make it through the "under cursor" check. 
Modes
Normal: This is the standard (classic) targeting mode, while active, your cursor doesn't filter out any unit types, and considers everything as potential valid targets.
Target Non-Hero Units Only: When this targeting mode is active, your cursor will ignore any Heroes or Structures as potentially valid targets.
Target Heroes Only: When this targeting mode is active your cursor will only consider Heroes as potentially valid targets and will ignore structures and all other types of units.
Toggle Commands
These commands will toggle the targeting modes on/off (making the targeting mode you pick, linger as long as you have it toggled on).
'Toggle Target Heroes Only' Command: Default Binding [Ctrl + H]
'Toggle Target Non-Hero Units Only' Command: Default Binding [Ctrl + G]
Hold Down Commands
These commands are only active when you're holding down the bound key, and they override the current "active mode". The only exception to this is if you hold down the key for the currently active mode, it will revert the mode back to "Normal" while the key is held down.
'Target Heroes Only' Command: Default Binding [H]
'Target Non-Hero Units Only' Command: Default Binding [G]

TL;DR: You can press a few keys to make your cursor only target heroes, so you'll never errantly click on a tower, minions, etc.

Thanks for the suggestion!

**Comment by xAzta** [comment]
Thanks :)
